livre/lib/data/BoundingAxis.cpp

```diff
@@ -0,0 +1,368 @@
+        _append3dVector( bbAxisData.normals2, normals[i] );
+        _append4dVector( bbAxisData.colors, color );
+        bbAxisData.types.push_back( BBOX );
+
+        ++_nVertices;
+    }
+
+    _createTicks( bbAxisData );
+    _sendDataToGPU( bbAxisData );
+}
+
+BoundingAxis::~BoundingAxis()
+{
+    glDeleteBuffers( 1, &_axisBufferId );
+    glDeleteBuffers( 1, &_typeBufferId );
+}
+
+void BoundingAxis::_sendDataToGPU( const BoundingAxisData& bbAxisData )
+{
+    glGenBuffers( 1, &_axisBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _axisBufferId );
+
+    size_t totalSize = sizeof(float) * _nVertices * 13;
+    glBufferData( GL_ARRAY_BUFFER, totalSize, nullptr, GL_STATIC_DRAW );
+
+    glBufferSubData( GL_ARRAY_BUFFER, 0, sizeof(float) * bbAxisData.vertices.size(),
+                     &bbAxisData.vertices[0] );
+    glBufferSubData( GL_ARRAY_BUFFER, sizeof(float) * ( 3 * _nVertices ),
+                     sizeof(float) * bbAxisData.normals.size(), &bbAxisData.normals[0] );
+    glBufferSubData( GL_ARRAY_BUFFER, sizeof(float) * ( 6 * _nVertices ),
+                     sizeof(float) * bbAxisData.normals2.size(), &bbAxisData.normals2[0] );
+    glBufferSubData( GL_ARRAY_BUFFER, sizeof(float) * ( 9 * _nVertices ),
+                     sizeof(float) * bbAxisData.colors.size(), &bbAxisData.colors[0] );
+
+    glGenBuffers( 1, &_typeBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLubyte) * bbAxisData.types.size(),
+                  &bbAxisData.types[0], GL_STATIC_DRAW );
+}
+
+void BoundingAxis::draw()
+{
+    glEnableVertexAttribArray( 0 );
+    glBindBuffer( GL_ARRAY_BUFFER, _axisBufferId );
+    glVertexAttribPointer( 0, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 1 );
+    glVertexAttribPointer( 1, 3, GL_FLOAT, GL_FALSE, 0,
+                           (GLvoid*)( sizeof(float) * ( 3 * _nVertices )));
+
+    glEnableVertexAttribArray( 2 );
+    glVertexAttribPointer( 2, 3, GL_FLOAT, GL_FALSE, 0,
+                           (GLvoid*)( sizeof(float) * ( 6 * _nVertices )));
+
+    glEnableVertexAttribArray( 3 );
+    glVertexAttribPointer( 3, 4, GL_FLOAT, GL_FALSE, 0,
+                           (GLvoid*)( sizeof(float) * ( 9 * _nVertices )));
+
+    glEnableVertexAttribArray( 4 );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glVertexAttribIPointer( 4, 1, GL_UNSIGNED_BYTE, 0, 0 );
+
+    glDrawArrays( GL_LINES, 0, _nVertices );
+
+    glDisableVertexAttribArray( 0 );
+    glDisableVertexAttribArray( 1 );
+    glDisableVertexAttribArray( 2 );
+    glDisableVertexAttribArray( 3 );
+}
+
+void BoundingAxis::_createTicks( BoundingAxisData& bbAxisData )
+{
+    const size_t longestBBAxis = bbAxisData.volInfo.worldSize.find_max_index();
+
+    const Vector2f range = _computeRange( bbAxisData.volInfo, longestBBAxis );
+
+    const uint32_t maxTickNumber = 50;
+    const float tickSizes[4] = { 1, 2, 5, 10 };
```

Comment:
This should be called tickDistances for consistency, shouldn't it? And the same goes for bbAxisData.tickSize.